XLS-0070d-credentials/README.md

```diff
@@ -0,0 +1,595 @@
+
+## 3. On-Ledger Object: `DepositPreauth`
+
+The `DepositPreauth` object tracks a preauthorization from one account to another. This object already exists on the XRPL, but is being extended as a part of this spec to also support credential preauthorization.
+
+### 3.1. Fields
+<details>
+<summary>
+
+As a reference, [here](https://xrpl.org/docs/references/protocol/ledger-data/ledger-entry-types/depositpreauth/#depositpreauth-fields) are the existing fields for the `DepositPreauth` object.
+</summary>
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`Account`|✔️|`string`|`AccountID`|The account that granted the preauthorization (the destination of the preauthorized payments).|
+|`Authorize`|✔️|`string`|`AccountID`|The account that received the preauthorization (the sender of the preauthorized payments).|
+|`LedgerEntryType`|✔️|`string`|`UInt16`|The value `0x0070`, mapped to the string `"DepositPreauth"`, indicates that this is a `DepositPreauth` object.|
+|`OwnerNode`|✔️|`string`|`UInt64`|A hint indicating which page of the sender's owner directory links to this object, in case the directory consists of multiple pages. Note: The object does not contain a direct link to the owner directory containing it, since that value can be derived from the `Account.PreviousTxnID`.|
+|`PreviousTxnID`|✔️|`string`|`Hash256`|The identifying hash of the transaction that most recently modified this object.|
+|`PreviousTxnLgrSeqNumber`|✔️|`number`|`UInt32`|The index of the ledger that contains the transaction that most recently modified this object.|
+</details>
+
+We propose these modifications:
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`Authorize`| |`string`|`AccountID`|This field already exists, but is becoming optional.|
+|`AuthorizeCredentials`| |`array`|`STArray`|The credential(s) that received the preauthorization. (Any account with these credentials can send preauthorized payments).|
+
+#### 3.1.1. Object ID
+
+The ID of this object will be either a hash of the `Account` and `Authorize` fields (as it currently is), or a hash of the `Account` and the contents of `AuthorizeCredentials` fields, combined with the unique space key for `DepositAuth` objects: `0x0070`.
+
+#### 3.1.2. `Authorize`
+
+This field is now optional, and a valid `DepositPreauth` object must have **exactly one of** the `Authorize` field or the `AuthorizeCredentials` field.
+
+#### 3.1.3. `AuthorizeCredentials`
+
+This field is an array of inner objects. The contents of these inner objects determine the credential(s) that are accepted.
+
+If more than one credential is included in the list, all of those credentials must be included (effectively ANDing them together).
+
+The list has a minimum size of 1 and a maximum size of 8 credentials.
```

Comment:
No, because those objects are useless unless you're actually using the Deposit Auth flow, and there isn't really an incentive for an account to create spam just for the sake of spamming the ledger.

Reply:
okay